### PR TITLE
Fix Review

### PR DESCRIPTION
**Pull request by sherlock-admin4**
Fix Review

### CODE REVIEW

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 1]
```diff
@@ -52,6 +59,7 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
     error InvalidEndBlock();
     error InvalidDeposit();
     error InvalidBooster();
+    error InvalidPointsPerBlock();
```
Comment (sherlock-admin2) [inline farming-contracts/contracts/farm/SophonFarming.sol:62]: Change related to https://github.com/sophon-org/farming-contracts/commit/f0b82fd83a5d85eb8dc7ba2bbce2d49fefb326a4. 

Fixes issues: [#23](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/23), [#79](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/79), [#181](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/181), [#118](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/118), [#27](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/27), [#89](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/89), [#185](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/185), [#154](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/154), [#36](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/36), [#13](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/13), [#39](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/39), [#112](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/112), [#18](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/18), [#224](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/224), [#180](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/180)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 2]
```diff
@@ -99,25 +107,29 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
...
      */
-    function initialize(uint256 ethAllocPoint_, uint256 sDAIAllocPoint_, uint256 _pointsPerBlock, uint256 _startBlock, uint256 _boosterMultiplier) public virtual onlyOwner {
+    function initialize(uint256 wstEthAllocPoint_, uint256 weEthAllocPoint_, uint256 sDAIAllocPoint_, uint256 _pointsPerBlock, uint256 _startBlock, uint256 _boosterMultiplier) public virtual onlyOwner {
         if (_initialized) {
             revert AlreadyInitialized();
         }
 
+        if (_pointsPerBlock < 1e18 || _pointsPerBlock > 1000e18) {
+            revert InvalidPointsPerBlock();
+        }
```
Comment (sherlock-admin3) [inline farming-contracts/contracts/farm/SophonFarming.sol:124]: Change related to https://github.com/sophon-org/farming-contracts/commit/f0b82fd83a5d85eb8dc7ba2bbce2d49fefb326a4. 

Fixes issues: [#23](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/23), [#79](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/79), [#181](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/181), [#118](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/118), [#27](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/27), [#89](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/89), [#185](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/185), [#154](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/154), [#36](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/36), [#13](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/13), [#39](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/39), [#112](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/112), [#18](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/18), [#224](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/224), [#180](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/180)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 3]
```diff
@@ -127,39 +139,41 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
         poolExists[stETH] = true;
         poolExists[eETH] = true;
 
+        _initialized = true;
+
         // sDAI
-        typeToId[PredefinedPool.sDAI] = add(sDAIAllocPoint_, sDAI, "sDAI", false);
+        typeToId[PredefinedPool.sDAI] = add(sDAIAllocPoint_, sDAI, "sDAI");
```
Comment (sherlock-admin4) [inline farming-contracts/contracts/farm/SophonFarming.sol:145]: Change related to https://github.com/sophon-org/farming-contracts/commit/6d117397d57a39a865cee0621cabf9154fb6fd71. 

Fixes issues: [#195](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/195), [#150](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/150), [#125](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/125), [#122](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/122), [#11](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/11), [#12](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/12), [#46](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/46), [#97](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/97), [#227](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/227), [#187](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/187), [#157](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/157), [#129](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/129), [#116](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/116), [#19](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/19), [#31](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/31), [#106](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/106), [#189](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/189)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 4]
```diff
@@ -127,39 +139,41 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
...
     /**
      * @notice Adds a new pool to the farm. Can only be called by the owner.
      * @param _allocPoint alloc point for new pool
      * @param _lpToken lpToken address
      * @param _description description of new pool
-     * @param _withUpdate True will update accounting for all pools
      * @return uint256 The pid of the newly created asset
      */
-    function add(uint256 _allocPoint, address _lpToken, string memory _description, bool _withUpdate) public onlyOwner returns (uint256) {
+    function add(uint256 _allocPoint, address _lpToken, string memory _description) public onlyOwner returns (uint256) {
```
Comment (sherlock-admin2) [inline farming-contracts/contracts/farm/SophonFarming.sol:164]: Change related to https://github.com/sophon-org/farming-contracts/commit/6d117397d57a39a865cee0621cabf9154fb6fd71. 

Fixes issues: [#195](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/195), [#150](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/150), [#125](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/125), [#122](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/122), [#11](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/11), [#12](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/12), [#46](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/46), [#97](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/97), [#227](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/227), [#187](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/187), [#157](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/157), [#129](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/129), [#116](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/116), [#19](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/19), [#31](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/31), [#106](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/106), [#189](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/189)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 5]
```diff
@@ -127,39 +139,41 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
...
         }
         if (isFarmingEnded()) {
             revert FarmingIsEnded();
         }
-        if (_withUpdate) {
-            massUpdatePools();
-        }
+
+        massUpdatePools();
+
```
Comment (sherlock-admin3) [inline farming-contracts/contracts/farm/SophonFarming.sol:176]: Change related to https://github.com/sophon-org/farming-contracts/commit/6d117397d57a39a865cee0621cabf9154fb6fd71. 

Fixes issues: [#195](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/195), [#150](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/150), [#125](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/125), [#122](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/122), [#11](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/11), [#12](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/12), [#46](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/46), [#97](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/97), [#227](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/227), [#187](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/187), [#157](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/157), [#129](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/129), [#116](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/116), [#19](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/19), [#31](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/31), [#106](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/106), [#189](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/189)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 6]
```diff
@@ -190,15 +204,13 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
...
-    function set(uint256 _pid, uint256 _allocPoint, bool _withUpdate) public onlyOwner {
+    function set(uint256 _pid, uint256 _allocPoint) external onlyOwner {
         if (isFarmingEnded()) {
             revert FarmingIsEnded();
         }
-        if (_withUpdate) {
-            massUpdatePools();
-        }
+
+        massUpdatePools();
```
Comment (sherlock-admin4) [inline farming-contracts/contracts/farm/SophonFarming.sol:213]: Change related to https://github.com/sophon-org/farming-contracts/commit/6d117397d57a39a865cee0621cabf9154fb6fd71. 

Fixes issues: [#195](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/195), [#150](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/150), [#125](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/125), [#122](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/122), [#11](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/11), [#12](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/12), [#46](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/46), [#97](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/97), [#227](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/227), [#187](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/187), [#157](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/157), [#129](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/129), [#116](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/116), [#19](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/19), [#31](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/31), [#106](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/106), [#189](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/189)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 7]
```diff
@@ -252,30 +264,42 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
...
 
     /**
      * @notice Set the start block of the farm
      * @param _startBlock the start block
      */
-    function setStartBlock(uint256 _startBlock) public onlyOwner {
-        if (_startBlock == 0 || (endBlock != 0 && _startBlock >= endBlock)) {
+    function setStartBlock(uint256 _startBlock) external onlyOwner {
+        uint256 blockNumber = getBlockNumber();
+        if (_startBlock == 0 || blockNumber > _startBlock || (endBlock != 0 && _startBlock >= endBlock)) {
```
Comment (sherlock-admin2) [inline farming-contracts/contracts/farm/SophonFarming.sol:295]: Change related to https://github.com/sophon-org/farming-contracts/commit/1f8d4e5ccec63052fcaf751b867294472e2d20a7. 

Fixes issues: [#108](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/108), [#205](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/205), [#40](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/40), [#72](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/72)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 8]
```diff
@@ -252,30 +264,42 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
...
             revert InvalidStartBlock();
         }
-        if (getBlockNumber() > startBlock) {
+        if (blockNumber > startBlock) {
```
Comment (sherlock-admin3) [inline farming-contracts/contracts/farm/SophonFarming.sol:298]: Change related to https://github.com/sophon-org/farming-contracts/commit/1f8d4e5ccec63052fcaf751b867294472e2d20a7. 

Fixes issues: [#108](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/108), [#205](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/205), [#40](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/40), [#72](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/72)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 9]
```diff
@@ -252,30 +264,42 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
...
             revert FarmingIsStarted();
         }
+
+        massUpdatePools();
```
Comment (sherlock-admin4) [inline farming-contracts/contracts/farm/SophonFarming.sol:302]: Change related to https://github.com/sophon-org/farming-contracts/commit/1f8d4e5ccec63052fcaf751b867294472e2d20a7. 

Fixes issues: [#108](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/108), [#205](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/205), [#40](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/40), [#72](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/72)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 10]
```diff
@@ -307,10 +331,14 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
...
      */
-    function setPointsPerBlock(uint256 _pointsPerBlock) public onlyOwner {
+    function setPointsPerBlock(uint256 _pointsPerBlock) external onlyOwner {
         if (isFarmingEnded()) {
             revert FarmingIsEnded();
         }
+        if (_pointsPerBlock < 1e18 || _pointsPerBlock > 1000e18) {
+            revert InvalidPointsPerBlock();
+        }
+
```
Comment (sherlock-admin2) [inline farming-contracts/contracts/farm/SophonFarming.sol:341]: Change related to https://github.com/sophon-org/farming-contracts/commit/f0b82fd83a5d85eb8dc7ba2bbce2d49fefb326a4. 

Fixes issues: [#23](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/23), [#79](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/79), [#181](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/181), [#118](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/118), [#27](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/27), [#89](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/89), [#185](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/185), [#154](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/154), [#36](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/36), [#13](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/13), [#39](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/39), [#112](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/112), [#18](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/18), [#224](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/224), [#180](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/180)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 11]
```diff
@@ -806,10 +843,9 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
      * @return uint256 out amount
      */
     function _ethTOstEth(uint256 _amount) internal returns (uint256) {
-        // submit function does not return exact amount of stETH so we need to check balances
-        uint256 balanceBefore = IERC20(stETH).balanceOf(address(this));
-        IstETH(stETH).submit{value: _amount}(address(this));
-        return (IERC20(stETH).balanceOf(address(this)) - balanceBefore);
+        return IstETH(stETH).getPooledEthByShares(
+            IstETH(stETH).submit{value: _amount}(owner())
+        );
```
Comment (sherlock-admin3) [inline farming-contracts/contracts/farm/SophonFarming.sol:848]: Change related to https://github.com/sophon-org/farming-contracts/commit/73adb6759b899e1a192fb5e28a5cd6202b5c3ff2. 

Fixes issues: [#64](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/64), [#186](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/186), [#4](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/4), [#92](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/92), [#30](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/30), [#119](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/119), [#102](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/102)

File: farming-contracts/contracts/farm/SophonFarming.sol [hunk 12]
```diff
@@ -830,8 +866,9 @@ contract SophonFarming is Upgradeable2Step, SophonFarmingState {
      * @return uint256 out amount
      */
     function _ethTOeEth(uint256 _amount) internal returns (uint256) {
-        // deposit returns exact amount of eETH
-        return IeETHLiquidityPool(eETHLiquidityPool).deposit{value: _amount}(address(this));
+        return IeETHLiquidityPool(eETHLiquidityPool).amountForShare(
+            IeETHLiquidityPool(eETHLiquidityPool).deposit{value: _amount}(owner())
+        );
```
Comment (sherlock-admin4) [inline farming-contracts/contracts/farm/SophonFarming.sol:871]: Change related to https://github.com/sophon-org/farming-contracts/commit/73adb6759b899e1a192fb5e28a5cd6202b5c3ff2. 

Fixes issues: [#64](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/64), [#186](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/186), [#4](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/4), [#92](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/92), [#30](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/30), [#119](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/119), [#102](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/102)

File: farming-contracts/contracts/farm/interfaces/IeETHLiquidityPool.sol [hunk 13]
```diff
@@ -4,4 +4,6 @@ pragma solidity 0.8.25;
 
 interface IeETHLiquidityPool {
     function deposit(address _referral) external payable returns (uint256);
+    function sharesForAmount(uint256 _amount) external view returns (uint256);
+    function amountForShare(uint256 _share) external view returns (uint256);
```
Comment (sherlock-admin2) [inline farming-contracts/contracts/farm/interfaces/IeETHLiquidityPool.sol:8]: Change related to https://github.com/sophon-org/farming-contracts/commit/73adb6759b899e1a192fb5e28a5cd6202b5c3ff2. 

Fixes issues: [#64](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/64), [#186](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/186), [#4](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/4), [#92](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/92), [#30](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/30), [#119](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/119), [#102](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/102)

File: farming-contracts/contracts/farm/interfaces/IstETH.sol [hunk 14]
```diff
@@ -4,4 +4,6 @@ pragma solidity 0.8.25;
 
 interface IstETH {
     function submit(address _referral) external payable returns (uint256);
+    function getSharesByPooledEth(uint256 _ethAmount) external view returns (uint256);
+    function getPooledEthByShares(uint256 _sharesAmount) external view returns (uint256);
```
Comment (sherlock-admin3) [inline farming-contracts/contracts/farm/interfaces/IstETH.sol:8]: Change related to https://github.com/sophon-org/farming-contracts/commit/73adb6759b899e1a192fb5e28a5cd6202b5c3ff2. 

Fixes issues: [#64](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/64), [#186](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/186), [#4](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/4), [#92](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/92), [#30](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/30), [#119](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/119), [#102](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/102)

File: farming-contracts/contracts/mocks/MockStETH.sol [hunk 15]
```diff
@@ -38,6 +38,14 @@ contract MockStETH is MockERC20, IstETH {
...
     }
 
+    function getSharesByPooledEth(uint256 _ethAmount) external view returns (uint256) {
+        return _ethAmount;
+    }
+
+    function getPooledEthByShares(uint256 _sharesAmount) external view returns (uint256) {
+        return _sharesAmount;
+    }
+
```
Comment (sherlock-admin4) [inline farming-contracts/contracts/mocks/MockStETH.sol:48]: Change related to https://github.com/sophon-org/farming-contracts/commit/73adb6759b899e1a192fb5e28a5cd6202b5c3ff2. 

Fixes issues: [#64](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/64), [#186](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/186), [#4](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/4), [#92](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/92), [#30](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/30), [#119](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/119), [#102](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/102)

File: farming-contracts/contracts/mocks/MockeETHLiquidityPool.sol [hunk 16]
```diff
@@ -37,4 +37,12 @@ contract MockeETHLiquidityPool is IeETHLiquidityPool {
...
         return returnAmount;
     }
+
+    function sharesForAmount(uint256 _amount) external view returns (uint256) {
+        return _amount;
+    }
+
+    function amountForShare(uint256 _share) external view returns (uint256) {
+        return _share;
+    }
```
Comment (sherlock-admin2) [inline farming-contracts/contracts/mocks/MockeETHLiquidityPool.sol:47]: Change related to https://github.com/sophon-org/farming-contracts/commit/73adb6759b899e1a192fb5e28a5cd6202b5c3ff2. 

Fixes issues: [#64](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/64), [#186](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/186), [#4](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/4), [#92](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/92), [#30](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/30), [#119](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/119), [#102](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/102)

File: farming-contracts/scripts/deploy.py [hunk 17]
```diff
@@ -148,12 +149,12 @@ def createMockSetup(deployTokens = False):
...
 
-    createFarm(weth, stETH, wstETH, wstETHAllocPoint, eETH, eETHLiquidityPool, weETH, dai, sDAI, sDAIAllocPoint, pointsPerBlock, startBlock, boosterMultiplier)
+    createFarm(weth, stETH, wstETH, wstEthAllocPoint, eETH, eETHLiquidityPool, weETH, weEthAllocPoint, dai, sDAI, sDAIAllocPoint, pointsPerBlock, startBlock, boosterMultiplier)
 
     acct, acct1, acct2, farm, mock0, mock1, weth, stETH, wstETH, eETH, eETHLiquidityPool, weETH, dai, sDAI = getMocks()
 
-    farm.add(10000, mock0, "mock0", True, {"from": acct})
-    farm.add(30000, mock1, "mock1", True, {"from": acct})
+    farm.add(10000, mock0, "mock0", {"from": acct})
+    farm.add(30000, mock1, "mock1", {"from": acct})
```
Comment (sherlock-admin3) [inline farming-contracts/scripts/deploy.py:157]: Change related to https://github.com/sophon-org/farming-contracts/commit/6d117397d57a39a865cee0621cabf9154fb6fd71. 

Fixes issues: [#195](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/195), [#150](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/150), [#125](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/125), [#122](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/122), [#11](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/11), [#12](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/12), [#46](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/46), [#97](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/97), [#227](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/227), [#187](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/187), [#157](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/157), [#129](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/129), [#116](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/116), [#19](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/19), [#31](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/31), [#106](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/106), [#189](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/189)

File: farming-contracts/scripts/deploy.py [hunk 18]
```diff
@@ -253,18 +254,19 @@ def testMainnetOnFork():
...
```
Comment (sherlock-admin4) [inline farming-contracts/scripts/deploy.py:269]: Change related to https://github.com/sophon-org/farming-contracts/commit/6d117397d57a39a865cee0621cabf9154fb6fd71. 

Fixes issues: [#195](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/195), [#150](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/150), [#125](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/125), [#122](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/122), [#11](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/11), [#12](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/12), [#46](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/46), [#97](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/97), [#227](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/227), [#187](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/187), [#157](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/157), [#129](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/129), [#116](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/116), [#19](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/19), [#31](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/31), [#106](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/106), [#189](https://github.com/sherlock-audit/2024-05-sophon-judging/issues/189)

**Pull request closed without merging**